test/CoreTest.cpp

```diff
@@ -71,6 +81,37 @@ TEST_CASE( "attribute_dtype_test", "[core]" )
     REQUIRE(Datatype::ARR_DBL_7 == a.dtype);
     a = Attribute(static_cast< bool >(false));
     REQUIRE(Datatype::BOOL == a.dtype);
+    
+    // fixed size integers
+    a = Attribute(static_cast< int16_t >(0));
+    REQUIRE(determineDatatype< int16_t >() == a.dtype);
+    a = Attribute(static_cast< int32_t >(0));
+    REQUIRE(determineDatatype< int32_t >() == a.dtype);
+    a = Attribute(static_cast< int64_t >(0));
+    REQUIRE(determineDatatype< int64_t >() == a.dtype);
+    a = Attribute(static_cast< uint16_t >(0));
+    REQUIRE(determineDatatype< uint16_t >() == a.dtype);
+    a = Attribute(static_cast< uint32_t >(0));
+    REQUIRE(determineDatatype< uint32_t >() == a.dtype);
+    a = Attribute(static_cast< uint64_t >(0));
+    REQUIRE(determineDatatype< uint64_t >() == a.dtype);
+    // TODO fixed size floats
+
+    // same implementation types (not necessary aliases) detection
+    if( sizeof(long) == sizeof(long long) )
+    {
+        a = Attribute(static_cast< long >(0));
+        REQUIRE(Datatype::LONGLONG == a.dtype);
+        a = Attribute(static_cast< long long >(0));
+        REQUIRE(Datatype::LONG == a.dtype);
+    }
+    if( sizeof(int) == sizeof(long) )
+    {
+        a = Attribute(static_cast< long >(0));
+        REQUIRE(Datatype::INT == a.dtype);
+        a = Attribute(static_cast< int >(0));
+        REQUIRE(Datatype::LONG == a.dtype);
+    }
```

Comment:
Scary thought:
There might be platforms where 
```cpp
sizeof(short) > 2u
```
making it impossible to write or read 16 Bit wide data (as `int16_t` does not alias to any of our provided integer types).

Reply:
Hm, that would be weird. But if a platform does not implement a 2 byte fundamental integer type, there will also be no alias `int16_t` for it.

Reply:
Just to throw out [this](https://stackoverflow.com/questions/5516044/system-where-1-byte-8-bit/5516161#5516161) SO answer again, this is present for `char` on certain (albeit rare) systems.

Reply:
:D